### PR TITLE
User can view statuses of Users.

### PR DESCRIPTION
**Pull request by kennykimjr**
usersearch

### CODE REVIEW

File: public/utility.js [hunk 1]
```diff
@@ -143,3 +150,64 @@ function safeAPIRequest(verb, route, token) {
     return response.json()
   })
 }
+
+function renderValue(property, value) {
+  const mapped = createElement('div')
```
Comment (thebearingedge) [inline public/utility.js]: Stick with the `$` prefix naming convention for elements. Review your codebase for consistency.

File: public/utility.js [hunk 2]
```diff
@@ -143,3 +150,64 @@ function safeAPIRequest(verb, route, token) {
...
+  else if (property === 'joined_at') {
+    mapped.textContent = new Date(value).toUTCString()
+  }
+  else {
+    mapped.textContent = value
+  }
+  return mapped
+}
+
+function renderRoles(target, roles) {
```
Comment (thebearingedge) [inline public/utility.js]: You shouldn't be passing a `target`. Instead, return a value from this function and do the appending at the callsite.

File: public/utility.js [hunk 3]
```diff
@@ -143,3 +150,64 @@ function safeAPIRequest(verb, route, token) {
...
+    }
+    else if (typeof object === 'string') {
+      newRole = storedServer.roles.find( (role) => { return role.id === object })
+    }
+    const newRoleElem = createElement('a', {'target':'_blank','textContent': newRole.name, 'href':'https://discordapi.com/permissions.html#' + newRole.permissions, 'style':'display: block'})
+    target.appendChild(newRoleElem)
+  })
+}
+
+function renderUser(target, userObject) {
```
Comment (thebearingedge) [inline public/utility.js]: You shouldn't be passing a `target`. Instead, return a value from this function and do the appending at the callsite.

File: public/utility.js [hunk 4]
```diff
@@ -143,3 +150,64 @@ function safeAPIRequest(verb, route, token) {
...
+  for (var property in userObject) {
+    const newProperty = createElement('div', {'textContent': property + ': ' + userObject[property] })
+    target.appendChild(newProperty)
+  }
+  return target
+}
+
+function renderTwoTable(table, obj) {
```
Comment (thebearingedge) [inline public/utility.js]: You shouldn't be passing a `table`. Instead, return a value from this function and do the appending at the call site.

File: public/utility.js [hunk 5]
```diff
@@ -143,3 +150,64 @@ function safeAPIRequest(verb, route, token) {
...
+  clearChildren(table)
+  for (var prop in obj) {
+    const mapped = renderValue(prop, obj[prop])
+    const newRow = createElement('div', {'className': 'row'})
+    const field = createElement('div', {'className':'column', 'textContent': prop.replace('_', ' '), 'style':'color: white'})
+    const value = createElement('div', {'className':'column', 'style': 'color: white'})
+    value.appendChild(mapped)
```
Comment (thebearingedge) [inline public/utility.js]: Extend your `createElement` function to append children for you.

File: public/utility.js [hunk 6]
```diff
@@ -143,3 +150,64 @@ function safeAPIRequest(verb, route, token) {
...
+function renderUser(target, userObject) {
+  for (var property in userObject) {
+    const newProperty = createElement('div', {'textContent': property + ': ' + userObject[property] })
+    target.appendChild(newProperty)
+  }
+  return target
+}
+
+function renderTwoTable(table, obj) {
+  clearChildren(table)
```
Comment (thebearingedge) [inline public/utility.js]: Clear the table before your call site.

File: public/utility.js [hunk 7]
```diff
@@ -143,3 +150,64 @@ function safeAPIRequest(verb, route, token) {
...
+    else if (typeof object === 'string') {
+      newRole = storedServer.roles.find( (role) => { return role.id === object })
+    }
+    const newRoleElem = createElement('a', {'target':'_blank','textContent': newRole.name, 'href':'https://discordapi.com/permissions.html#' + newRole.permissions, 'style':'display: block'})
+    target.appendChild(newRoleElem)
+  })
+}
+
+function renderUser(target, userObject) {
+  for (var property in userObject) {
```
Comment (thebearingedge) [inline public/utility.js]: You can just call this `user`. Try not to put "coding" terms in your business logic. Review your codebase for consistency.

File: public/utility.js [hunk 8]
```diff
@@ -143,3 +150,64 @@ function safeAPIRequest(verb, route, token) {
...
+    mapped.textContent = value
+  }
+  return mapped
+}
+
+function renderRoles(target, roles) {
+  roles.forEach(object => {
+    let newRole = null
+    if (typeof object === 'object') {
+      newRole = storedServer.roles.find( (role) => { return role.id === object.id })
```
Comment (thebearingedge) [inline public/utility.js]: You have extra whitespace before your arrow function parameter. Remove it and review your codebase for consistency.

Also, arrow functions implicitly return, if its body only contains one statement, omit the curly braces and `return` keyword. Review your codebase for consistency.

File: public/utility.js [hunk 9]
```diff
@@ -143,3 +150,64 @@ function safeAPIRequest(verb, route, token) {
     return response.json()
   })
 }
+
+function renderValue(property, value) {
+  const mapped = createElement('div')
+  if (property === 'emojis') {
+    value.forEach(object => {
+      const newObject = createElement('div', {'textContent': object.name})
```
Comment (thebearingedge) [inline public/utility.js]: `newObject` is a bad name. What does this represent in your business domain? Name it that. Review your codebase for vague names.

File: public/utility.js [hunk 10]
```diff
@@ -18,8 +18,15 @@ const $channelstats = document.getElementById('channelstats')
 const $serversearchicon = document.getElementById('server-search-icon')
 const $servername = document.getElementById('servername')
 const $userstats = document.getElementById('userstats')
+const $usersearchicon = document.getElementById('user-search-icon')
```
Comment (thebearingedge) [inline public/utility.js]: Use `camelCase` for JavaScript variables. Review your codebase for consistency.

File: public/styles.css [hunk 11]
```diff
@@ -243,20 +243,68 @@ hr {
...
-#serversearchicon {
+#server-search-icon,
+#user-search-icon {
   margin: 0 auto;
   display: block;
   border-radius: 50%;
 }
 
-#servername {
+#servername,
```
Comment (thebearingedge) [inline public/styles.css]: Use `kebab-case` for CSS selectors and HTML attributes. Review your codebase for consistency.

File: public/index.js [hunk 12]
```diff
@@ -126,27 +126,59 @@ $serverstats.addEventListener('click', (event) => {
...
     .catch(error => {
       alert(error.message)
     })
   })
 })
+
+$userstats.addEventListener('click', (event) => {
+  if (event.target.id === 'refresh-users') {
+    clearChildren($usersearch)
+    safeToken()
```
Comment (thebearingedge) [inline public/index.js]: Extract this routine into its own function. Keep your event listeners lean.

**Comment by thebearingedge** [comment]
Please reopen when ready.

**Comment by kennykimjr** [comment]
screen shot 2017-07-18 at 8 55 59 pm

**Comment by kennykimjr** [comment]
screen shot 2017-07-19 at 1 36 52 pm

**Pull request merged**